### DISCUSSION TITLE
oneOf matcher does not work for property with data types string | empty (“”) | null

### DISCUSSION THREAD

**Discussion by asub927** [pactumjs/pactum#267] [Q&A]
I need to validate an api which has a response property which returns string or empty or null values based on the data in the backend. 
- Initially tried with string() matcher which fails when the property value returns empty or null
- then tried oneOf() matcher which also fails when the property value returns string()
- works when I use any() but I want to match specific to the allowed data types

Code sample for reference. Any help to handle this condition is greatly appreciated. 

_spec
.get(/api)
.expectStatus(200)
.expectJsonMatch({
property: oneOf([string(), "", null])}

**Comment by ASaiAnudeep** [comment]
With the current implementation of [oneOf](https://pactumjs.github.io/api/matching/oneOf.html), it only supports strings. Ex:

```js
{ "gender": oneOf(["male", "female"]) }
```

Marking this as a feature request.

For a workaround, please use [expression](https://pactumjs.github.io/api/matching/expression.html) matcher.
